### PR TITLE
[BI-692] - Automatically set scale class to `Numerical` with `Computation` method class

### PR DESCRIPTION
**Pull request by ctucker3**

### CODE REVIEW

File: src/main/java/org/breedinginsight/services/TraitService.java [hunk 1]
```diff
@@ -269,4 +273,18 @@ public void assignTraitsProgramObservationLevel(List<Trait> traits, UUID program
...
+
+    public void preprocessTraits(List<Trait> traits) {
+
+        // Set data type to numerical when method class is computation
+        for (Trait trait: traits) {
+            if (trait.getMethod() != null && trait.getMethod().getMethodClass() != null &&
+                trait.getMethod().getMethodClass().equalsIgnoreCase(Method.COMPUTATION_TYPE)) {
+                if (trait.getScale() != null) {
+                    trait.getScale().setDataType(DataType.NUMERICAL);
+                }
```
Comment (timparsons) [inline src/main/java/org/breedinginsight/services/TraitService.java:285]: I thought we had agreed that the scale would always be set to `NUMERICAL`, even if there was a value previously set?

**Review by timparsons — CHANGES_REQUESTED** [review]

File: src/main/java/org/breedinginsight/services/TraitService.java (hunk 1 above)
Comment (ctucker3) [reply src/main/java/org/breedinginsight/services/TraitService.java:285]: It does do that. The only case where it wouldn't set the class to numerical is when no `Scale` is passed in, which would be caught by the checks after this section. If the user doesn't pass in a scale I want them to get an error that they didn't pass in a scale. If I'm not properly seeing what you are referencing, I'm open to a chat!

Comment (timparsons) [reply src/main/java/org/breedinginsight/services/TraitService.java:285]: Ahh! I misread this code yesterday.  Thanks!

**Review by timparsons — APPROVED** [review]

**Review by nickpalladino — CHANGES_REQUESTED** [review]
Could you add a test case for this?

**Comment by ctucker3** [comment]
> Could you add a test case for this?

Tests added

**Review by nickpalladino — APPROVED** [review]

**Pull request merged**